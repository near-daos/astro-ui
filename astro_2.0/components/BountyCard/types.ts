import { Token } from 'types/token';
import { BountyStatus } from 'types/bounties';

export enum CardType {
  Bounty = 'Bounty',
  Claim = 'Claim',
}
export interface BountyCardContent {
  id: string;
  daoId: string;
  amount: string;
  type: CardType;
  description: string;
  externalUrl?: string;
  token: Token;
  status: BountyStatus;
  bountyBond: string;
  slots?: number;
  slotsTotal?: number;
  claimedByCurrentUser: boolean;
  accountId?: string;
  timeToComplete: string;
  forgivenessPeriod: string;
  relatedProposalHash?: string;
}
