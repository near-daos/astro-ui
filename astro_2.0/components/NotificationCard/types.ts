import { NotificationType, NotificationStatus } from 'types/notification';

export interface NotificationCardContent {
  id: string;
  type: NotificationType;
  status: NotificationStatus;
  text: string;
  time: string;
  flagCover?: string;
  logo?: string;
  url?: string;
}
