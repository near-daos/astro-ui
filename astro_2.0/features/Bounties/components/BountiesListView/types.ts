import { Bounty } from 'types/bounties';
import { ReactNode } from 'react';
import { ProposalVariant } from 'types/proposal';

export type SectionItem = {
  id: string;
  title: string;
  proposer: string;
  proposalId: string;
  bounty?: Bounty;
  link: {
    pathname: string;
    query: {
      dao: string;
      bounty?: string;
      proposal?: string;
    };
  };
  completeHandler?: (
    id: number,
    variant: ProposalVariant.ProposeDoneBounty
  ) => void;
  content: ReactNode;
};
