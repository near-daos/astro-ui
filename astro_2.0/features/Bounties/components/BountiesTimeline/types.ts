export type TimelineMilestoneType =
  | 'Proposal Created'
  | 'Bounty Created'
  | 'Claim'
  | 'Pending Approval'
  | 'Complete Claim'
  | 'Claim Deadline'
  | 'Complete Bounty';

export type TimelineMilestone = {
  type: TimelineMilestoneType;
  date: Date;
  tooltip: string;
  color?: string;
};

export type TimelineClaim = {
  id: string;
  milestones: TimelineMilestone[];
  title: string;
  color: string;
  minDate?: Date;
  maxDate?: Date;
};

export type TimelineGroup = {
  id: string;
  name: string;
  isOpen: boolean;
  milestones: TimelineMilestone[];
  claims: TimelineClaim[];
  minDate?: Date;
  maxDate?: Date;
};

export type TimelineGranularity = 'month' | 'day' | 'hour';
