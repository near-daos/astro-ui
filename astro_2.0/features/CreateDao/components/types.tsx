import { IconName } from 'components/Icon';

export type DAOType = 'club' | 'cooperative' | 'corporation' | 'foundation';
export type Subject = 'proposals' | 'structure' | 'voting';
export type DAOVotingPowerType = 'democratic' | 'weighted';
export type DAOProposalsType = 'open' | 'closed';
export type DAOStructureType = 'flat' | 'groups';
export type DaoImageType = 'flagCover' | 'flagLogo' | 'tokenImage';

export interface DAOTemplate {
  title: string;
  description: string;
  variant: DAOType;
  proposals: DAOProposalsType;
  structure: DAOStructureType;
  voting: DAOVotingPowerType;
  disabled?: boolean;
}

export type DaoSettingOption<T> = {
  title: string;
  value: T;
  icon: IconName;
  subject: Subject;
  description: string;
  disabled?: boolean;
};

export interface IDaoCreateForm {
  address: string;
  displayName: string;
  purpose: string;
  websites: string[];
  flagCover: FileList;
  flagLogo: FileList;
  defaultFlag: string;
}

export interface DAOFormValues extends IDaoCreateForm {
  proposals: DAOProposalsType;
  structure: DAOStructureType;
  voting: DAOVotingPowerType;
  legalStatus?: string;
  legalLink?: string;
  gas: number;
  members: string[];
}

export type DaoSubjectInfo = {
  subject: Subject;
  title: string;
  subTitle: string;
};
