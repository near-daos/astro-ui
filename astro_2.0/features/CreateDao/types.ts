import { SelectorRow } from 'astro_2.0/features/pages/nestedDaoPagesContent/DaoPolicyPageContent/helpers';

export type InfoStep = {
  displayName: string;
  address: string;
  purpose: string;
  isValid: boolean;
};

export type LegalStep = {
  legalStatus: string;
  legalLink: string;
  isValid: boolean;
};

export type ProposalsStep = {
  data: SelectorRow[] | null;
  isValid: boolean;
};

export type VotingStep = {
  data: SelectorRow[] | null;
  isValid: boolean;
};

export type SubmitStep = {
  gas: number | string;
  isValid: boolean;
};

export type LinksStep = {
  websites: string[];
  isValid: boolean;
};

export type MembersStep = {
  accounts: { name: string; role: string }[];
  isValid: boolean;
};

export type GroupsStep = {
  items: { name: string; slug?: string }[];
  isValid: boolean;
};

export type AssetsStep = {
  flagCover: string;
  flagLogo: string;
  defaultFlag: string;
  isValid: boolean;
};

declare module 'little-state-machine' {
  interface GlobalState
    extends Record<
      string,
      | InfoStep
      | LegalStep
      | LinksStep
      | GroupsStep
      | MembersStep
      | AssetsStep
      | ProposalsStep
      | VotingStep
      | SubmitStep
    > {
    info: InfoStep;
    kyc: LegalStep;
    links: LinksStep;
    members: MembersStep;
    groups: GroupsStep;
    assets: AssetsStep;
    submit: SubmitStep;
    proposals: ProposalsStep;
    voting: VotingStep;
  }
}
