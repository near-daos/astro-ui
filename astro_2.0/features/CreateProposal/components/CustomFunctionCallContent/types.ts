export enum FunctionCallType {
  Custom = 'Custom',
  BuyNFTfromMintbase = 'BuyNFTfromMintbase',
  TransferNFTfromMintbase = 'TransferNFTfromMintbase',
  MintNFTsfromMintbase = 'MintNFTsfromMintbase',
  BuyNFTfromParas = 'BuyNFTfromParas',
  SwapsOnRef = 'SwapsOnRef',
  VoteInAnotherDao = 'VoteInAnotherDao',
  RemoveUpgradeCode = 'RemoveUpgradeCode',
  CreateRoketoStream = 'CreateRoketoStream',
}
