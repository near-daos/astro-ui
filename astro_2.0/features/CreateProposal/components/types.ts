export type CreateTransferInput = {
  token: string;
  amount: number;
  details: string;
  externalUrl: string;
  target: string;
  gas: number;
};
