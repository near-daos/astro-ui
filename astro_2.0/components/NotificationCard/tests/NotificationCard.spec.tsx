/* eslint-disable @typescript-eslint/ban-ts-comment */

import { render } from 'jest/testUtils';
import { useRouter } from 'next/router';
import { fireEvent } from '@testing-library/dom';

import { ProposalType } from 'types/proposal';
import { NotificationStatus, NotifiedActionType } from 'types/notification';

import { getNotificationParamsByType } from 'astro_2.0/features/Notifications';

import {
  NotificationCard,
  NotificationCardProps,
} from 'astro_2.0/components/NotificationCard';

jest.mock('next/router', () => {
  return {
    useRouter: jest.fn(),
  };
});

jest.mock('next-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string): string => str,
    };
  },
}));

jest.mock('astro_2.0/features/Notifications', () => {
  return {
    ...jest.requireActual('astro_2.0/features/Notifications'),
    useNotifications: jest.fn(() => ({})),
    getNotificationParamsByType: jest.fn(() => ({})),
  };
});

describe('Notification Card', () => {
  function getNotificationComponent(props?: Partial<NotificationCardProps>) {
    return render(
      <NotificationCard
        id="1"
        accountId="2"
        isNew
        isRead={false}
        isMuted={false}
        isArchived={false}
        daoId="3"
        dao={null}
        signerId="4"
        targetId="5"
        type={NotifiedActionType.Vote}
        isMuteAvailable
        isMarkReadAvailable
        isDeleteAvailable
        status={NotificationStatus.Created}
        metadata={{
          args: '',
          methodName: 'act_proposal',
          proposal: {
            id: '6',
            proposer: 'proposer',
            description: 'some description',
            votes: {},
            kind: { type: ProposalType.Vote },
          },
        }}
        createdAt="2016-01-01"
        {...props}
      />
    );
  }

  it('Should render component', () => {
    const { container } = getNotificationComponent({
      metadata: {
        args: '',
        methodName: 'act_proposal',
        // @ts-ignore
        proposal: {
          id: '6',
          description: 'some description',
          votes: {},
          kind: { type: ProposalType.Vote },
        },
      },
    });

    expect(container).toMatchSnapshot();
  });

  describe('handleMarkReadClick', () => {
    it('Should call onMarkRead callback', () => {
      const id = '123';
      const onUpdate = jest.fn();

      const { getAllByTestId } = getNotificationComponent({ id, onUpdate });

      fireEvent.click(getAllByTestId('mark-read')[0]);

      expect(onUpdate).toBeCalledWith(id, {
        isArchived: false,
        isMuted: false,
        isRead: true,
      });
    });

    it('Should not fail if onMarkRead callback is not provided', () => {
      const { getAllByTestId } = getNotificationComponent();

      fireEvent.click(getAllByTestId('mark-read')[0]);
    });

    it('Should call handleUpdate with proper args', () => {
      const id = '123';
      const isMuted = true;
      const isArchived = true;
      const onUpdate = jest.fn();

      const { getAllByTestId } = getNotificationComponent({
        id,
        isMuted,
        isArchived,
        onUpdate,
      });

      fireEvent.click(getAllByTestId('mark-read')[0]);

      expect(onUpdate).toBeCalledWith(id, {
        isMuted,
        isArchived,
        isRead: true,
      });
    });
  });

  describe('handleDeleteClick', () => {
    it('Should call onRemove callback', () => {
      const id = '123';
      const onRemove = jest.fn();

      const { getAllByTestId } = getNotificationComponent({
        id,
        regular: false,
        onRemove,
      });

      fireEvent.click(getAllByTestId('delete-action-button')[0]);

      expect(onRemove).toBeCalledWith(id, {
        isArchived: true,
        isMuted: false,
        isRead: false,
      });
    });

    it('Should not fail if onRemove callback is not provided', () => {
      const { getAllByTestId } = getNotificationComponent({ regular: false });

      fireEvent.click(getAllByTestId('delete-action-button')[0]);
    });

    it('Should call handleUpdate with proper args', () => {
      const id = '123';
      const isMuted = true;
      const isRead = true;
      const isArchived = false;

      const onRemove = jest.fn();

      const { getAllByTestId } = getNotificationComponent({
        id,
        isMuted,
        isRead,
        isArchived,
        regular: false,
        onRemove,
      });

      fireEvent.click(getAllByTestId('delete-action-button')[0]);

      expect(onRemove).toBeCalledWith(id, {
        isMuted,
        isRead,
        isArchived: !isArchived,
      });
    });
  });

  describe('handleNotificationClick', () => {
    it('Should mark noty as read', () => {
      const id = '123';
      const onUpdate = jest.fn();

      const { getByTestId } = getNotificationComponent({ id, onUpdate });

      fireEvent.click(getByTestId('noty-content'));

      expect(onUpdate).toBeCalled();
    });

    it('Should navigate to url if provided', () => {
      const url = 'HelloWorld';

      // @ts-ignore
      getNotificationParamsByType.mockImplementation(() => ({ url }));

      const router = {
        push: jest.fn(),
      };

      // @ts-ignore
      useRouter.mockImplementation(() => router);

      const { getByTestId } = getNotificationComponent();

      fireEvent.click(getByTestId('noty-content'));

      expect(router.push).toBeCalledWith(url);
    });
  });
});

/* eslint-enable @typescript-eslint/ban-ts-comment */
