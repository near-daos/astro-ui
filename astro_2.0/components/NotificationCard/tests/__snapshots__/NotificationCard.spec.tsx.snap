// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Card Should render component 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="content new"
      data-testid="noty-content"
      role="button"
      tabindex="0"
    >
      <div
        class="flagContainer"
      >
        <div
          class="flagWrapper"
        >
          <div
            class="flag"
          >
            <div
              class="root logo md"
            >
              <svg
                class="defaultLogo"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
            </div>
          </div>
          <div
            class="type"
          />
        </div>
      </div>
      <div
        class="text"
      />
      <div
        class="description"
      >
        <div
          class="descriptionHeader"
        >
          description
          :
        </div>
        <span>
          some description
        </span>
      </div>
      <div
        class="time"
      >
        12:00 am
      </div>
      <div
        class="control"
      >
        <div
          class=""
        >
          <div
            class="markRead"
          >
            <button
              class="btn transparent sizeBlock markReadButton"
              data-testid="mark-read"
              type="button"
            >
              <svg
                class="markReadIcon"
                width="24"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mobileActionsPanel"
    >
      <div
        class=""
      >
        <div
          class="markRead"
        >
          <button
            class="btn transparent sizeBlock markReadButton"
            data-testid="mark-read"
            type="button"
          >
            <svg
              class="markReadIcon"
              width="24"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
